EMCECS/presto-s3-connector
# Issue 85: Add gradle checkstyle to keep code format consistent and reformat source code

**Changelog description:**

1. add IntelliJ IDE code format config file 
2. add gradle checkstyle plugin config files
3. reformat `src/` **(both main and test, except files in src/test/resources)** to follow consistent code format and checkstyle.


**Purpose of the change:** Fixes #85 


**What the code does:**

1. add `config/checkstyle/intelij.xml` for the IntelliJ IDE to import as a code format scheme in `Settings -> Editor -> Code Style -> Import Scheme -> IntelliJ IDEA code style XML`. This file is obtained from (https://github.com/pravega/pravega/blob/a5f5d670b7f2340b8388a9a2b216aa7e5b4d7151/checkstyle/intelij.xml). To apply Reformat Code with this configured code style, right click on `src/`, reformat code, include all subdirectories.
2. add `checkstyle` plugin in `build.gradle`
3. add checkstyleToolVersion to `8.23` (same as pravega/pravega)
3. add and modify `config/checkstyle/checkstyle.xml`. This file is obtained and modified from (https://github.com/pravega/pravega/blob/a5f5d670b7f2340b8388a9a2b216aa7e5b4d7151/checkstyle/checkstyle.xml)
     - Excluded the following tags `ConstantName, StaticVariableName, ClassTypeParameterName, MethodTypeParameterName, LocalVariableName` due to too much inconsistenties across too many files.
4. checkstyle was employing rules on auto generated files such as  `build/generated-test-avro-java`. Fixed by modifying `config/checkstyle/suppressions.xml` to exclude all files in `build/generated-test-avro-java` 
5. The rest of the files which are all under `src/main` and `src/test` except files in `src/test/resources` have been reformatted according to code format and gradle checkstyle.

**How to verify it:** 
`./gradlew clean build` (if any checkstyle rule is violated, there will be an error on build. Otherwise, build succeeds)
`./gradlew test`
`./gradlew test -Pintegration` all successful. Code formatting and checkstyle are expected to have absolutely no change in behavior so builds and tests are expected to pass.

**Misc:**
How to set up `gradle checkstyle` on IntelliJ IDE
1. `Settings -> Plugins -> Search for CheckStyle-IDEA and install`
2. `Settings -> Tools -> Checkstyle`
    - set `checkstyle version` to `8.23`
    - set `scan scope` to `All sources (including tests)`
    - add configuration file, use a `local checkstyle file`, set to `config/checkstyle/checkstyle.xml`
    - apply and ok
3. On the very bottom of the IDE pane, click on `CheckStyle` with yellow pencil mark. Click on the `Rules` dropdown list, click on the repo checkstyle you set up in steps above, `check project` on the left side.
